docs/bots-guide.md

```diff
@@ -443,6 +443,103 @@ bot_handler.update_message(dict(
   development environment hosted on your computer, use
   `localhost:9991`
 
+### Writing tests for bots
+
+Bots in the [`python-zulip-api`](https://github.com/zulip/python-zulip-api/tree/master/zulip_bots/zulip_bots/bots)
+repository should have unittests. This section will explain the basics of writing a test suite for a bot.
+
+*Unittests for bots make heavy use of mocking. If you want to get comfortable with mocking, mocking strategies, etc.
+ you should check out our [mocking guide](testing-with-django.html#testing-with-mocks).*
+
+#### A simple example
+
+ Let's have a look at a simple test suite for the [`helloworld`](
+ https://github.com/zulip/python-zulip-api/tree/master/zulip_bots/zulip_bots/bots/helloworld) bot
+ (the actual test is written slightly more compact).
+
+     #!/usr/bin/env python
```

Comment:
I don't think our test files need shebang lines, since nobody is going to run them as a script.

Reply:
Fair point; running them as a script actually fails.